scripts/cubemx-importer-utils.stanza

```diff
@@ -0,0 +1,152 @@
+defpackage ocdb/scripts/cubemx-importer-utils:
+  import core
+  import collections
+  import jitx
+  import pcre
+
+; Algorithm to sanitize pin names.
+defn sanitize-pin-name (s:String) -> String:
+  var s*: String = s
+  val chars = " (/-"
+  for c in chars do:
+    match(index-of-char(s*, c)):
+      (i:Int): s* = s*[0 to i]
+      (_:?): false
+  s*
+
+defn get-numbers-in-string (s:String) -> Vector<[Int, Int]>:
+  val arr = Vector<[Int, Int]>()
+  var start = -1
+  var end = -1
+  for i in 0 to length(s) do :
+    if (digit?(s[i])) :
+      if (start == -1) :
+        start = i
+    else if (start >= 0) :
+      end = i
+      ;println(("6: Adding %_ from %_ to %_") % [s, start, end])
+      add(arr, [start, end])
+      start = -1
+      end = -1
+
+  if (start >= 0) :
+    end = length(s)
+    ;println(("7: Adding %_ from %_ to %_") % [s, start, end])
+    add(arr, [start, end])
+  arr
+
+public defn extract-pin-name (raw:String) -> String:
+  val ret1 = sanitize-pin-name(raw)
+  ;println(ret1)
+  ;handle special cases first
+  val arr = get-numbers-in-string(ret1)
+  val l = length(ret1)
+  if ( suffix?(ret1, "_C") or
+      suffix?(ret1, "_INN") or
+      suffix?(ret1, "_VINM") or
+      prefix?(ret1, "DSI_D") or
+      prefix?(ret1, "DSIHOST_D") or
+      prefix?(ret1, "DDR_DQS")) :
+    ; These have array length of 1
+    val [start, end] = arr[0]
+    val pref = ret1[0 to start]
+    val sufx = ret1[end to l]
+    val num = ret1[start to end]
+    val ret = to-string("%_%_[%_]" % [pref, sufx, num])
+    ;println("1: %_ ==> %_" % [raw, ret])
+    ret
+  else if ( prefix?(ret1, "UCPD1_DBCC")):
+    ; These have array length of 1
+    val [start, end] = arr[1]
+    val num = ret1[start to end]
+    val ret = to-string("UCPD1_DBCC[%_]" % [num])
+    ;println("2: %_ ==> %_" % [raw, ret])
+    ret
+  else if ( length(arr) == 0 or
+            prefix?(ret1, "VDD") or
```

Comment:
I would suggest to extract voltage prefixed in a container with an explicit name: `voltage-...`, and use this container here